packages/graph/hash_graph/lib/graph/src/store/crud.rs

```diff
@@ -0,0 +1,22 @@
+//! Store interface for CRUD operations.
+//!
+//! The traits defined in this module are used in [`Store`] to create, read, update, and delete
+//! entries. They form a unified access to the [`Store`], so it's possible to add operations to the
+//! [`Store`] without changing the [`Store`] implementation.
+
+use async_trait::async_trait;
+use error_stack::Result;
+
+use crate::store::{QueryError, Store};
+
+/// Read access to a [`Store`].
+#[async_trait]
+pub trait Read<I: Sync, T>: Store {
```

Comment:
Is there a reason why the T here isn't an associated type? I am not sure what it refers to looking at just the trait :)

Reply:
`T` is specifying in which type is queried. It's not possible to make it associated as this is difference for implementations with the same index, e.g. `Read<VersionedUri, PropertyType>` and `Read<VersionedUri, EntityType>`. It uniquely identifies an operation.